### PR TITLE
feat: Add 4.9.0 to Catalog for Main and Spring Boot

### PR DESCRIPTION
**Pull request by apupier**
wondering if we should wait for Camel Kamelet and Camel Quarkus aligned with that too.

belongs to #1695

### CODE REVIEW

**Comment by apupier** [comment]
a bunch of tests to adjust (and to verify it is right to adjust and not a regression):

```
[@kaoto/kaoto]: Tests:       9 failed, 1 skipped, 2 todo, 1325 passed, 1337 total
[@kaoto/kaoto]: Snapshots:   7 failed, 134 passed, 141 total
```

**Comment by oscerd** [comment]
Camel Kamelets 4.9.0 is on vote.

**Comment by lordrip** [comment]
There are snapshots broken due to new parameters in some EIPs and the dataformat test is broken probably due to a new one being included.

**Review by lordrip — APPROVED** [review]

File: packages/catalog-generator/scripts/build-catalogs.mjs [hunk 1]
```diff
@@ -14,14 +14,14 @@ const require = createRequire(import.meta.url);
  **/
 const CATALOGS = {
   // https://repo1.maven.org/maven2/org/apache/camel/camel/
-  Main: ['4.8.1', '4.4.4', '4.4.0.redhat-00046', '4.8.0.redhat-00017'],
+  Main: ['4.9.0', '4.4.4', '4.4.0.redhat-00046', '4.8.0.redhat-00017'],
   // https://repo1.maven.org/maven2/org/apache/camel/quarkus/camel-quarkus/
   Quarkus: ['3.16.0', '3.8.3', '3.8.0.redhat-00018', '3.15.0.redhat-00007'],
   // https://repo1.maven.org/maven2/org/apache/camel/springboot/spring-boot/
-  SpringBoot: ['4.8.1', '4.4.4', '4.4.0.redhat-00039', '4.8.0.redhat-00022'],
+  SpringBoot: ['4.9.0', '4.4.4', '4.4.0.redhat-00039', '4.8.0.redhat-00022'],
```
Comment (lhein) [inline packages/catalog-generator/scripts/build-catalogs.mjs:21]: 4.8.1 as LTS should stay and only be upgraded to 4.8.2. That also applies to Main

**Review by lhein — CHANGES_REQUESTED** [review]
see my comment

**Review by lhein — APPROVED** [review]

**Comment by lhein** [comment]
Quarkus version still to be changed.

**Pull request merged**